shared/oae/api/oae.api.widget.js

```diff
@@ -275,6 +278,11 @@ define(['exports', 'jquery', 'underscore', 'oae.api.config', 'oae.api.i18n', 'oa
             // The data-widget attribute is removed, to avoid the widget being rendered again
             $element.removeAttr('data-widget');
 
+            // Apply the widget data from the widget loading element on to the widget data for this
+            // widget
+            widgetData[widgetId] = widgetData[widgetId] || {};
+            $.extend(widgetData[widgetId], $element.data());
```

Comment:
Would you expect the widget data from the attributes to override the widgetData that's been explicitly passed in or the other way around? I think this is doing the overrides the opposite way of how I expected it to work. Regardless, it would be good to add a bit of documentation to the `loadWidgets` functions to explain how this works.